indexes/utxoindex/src/stores/indexed_utxos.rs

```diff
@@ -145,22 +175,24 @@ impl UtxoSetByScriptPublicKeyStoreReader for DbUtxoSetByScriptPublicKeyStore {
         for script_public_key in script_public_keys.iter() {
             let script_public_key_bucket = ScriptPublicKeyBucket::from(script_public_key);
             let utxos_by_script_public_keys_inner = CompactUtxoCollection::from_iter(
-                self.access
-                    .seek_iterator::<TransactionOutpoint, CompactUtxoEntry>(
-                        Some(script_public_key_bucket.as_ref()),
-                        None,
-                        usize::MAX,
-                        false,
-                    )
-                    .map(move |value| {
-                        let (k, v) = value.expect("expected `key: TransactionOutpoint`, `value: CompactUtxoEntry`");
-                        (k, v)
-                    }),
+                self.access.seek_iterator(Some(script_public_key_bucket.as_ref()), None, usize::MAX, false).map(|res| {
+                    let (key, entry) = res.unwrap();
+                    let (_, outpoint) = UtxoEntryFullAccessKey(Arc::new(key.to_vec())).extract_script_public_key_and_outpoint();
```

Comment:
Using an extract_outpoint function can be cheaper, no need to deserialize the script_public_key.

Can just take self.0[(self.0.len() - TRANSACTION_OUTPOINT_KEY_SIZE)..]

Reply:
Also, the key bytes you retrieve from the seek iterator consists of transaction outpoint bytes (as it slices out the length of the bucket, i.e. the script_public_key bytes), as such the resulting key can simply be converted to an outpoint directly, there is no need for the UtxoEntryFullAccessKey here.